File: ValidBraces.py
from collections import deque

def validBraces(string):
    stack = deque()
    
    for c in string:
        if c == '(' or c=='[' or c =='{':
            stack.append(c)
        elif c == ')' and stack and stack[-1] == '(':
            stack.pop()
        elif c == ']' and stack and stack[-1] == '[':
            stack.pop()
        elif c == '}' and stack and stack[-1] == '{':
            stack.pop()
        else:
            return False
    if stack: return False
    else: return True

File: ValidBraces.java
import java.util.*;

public class BraceChecker {

  public boolean isValid(String braces) {
    Stack stack = new Stack();
    char[] bracesArray = braces.toCharArray();
    for (char c : bracesArray) {
      switch(c) {
          case '(':
          case '[':
          case '{':
            stack.push(c);
            break;
          case ')':
            if (!stack.empty() && stack.peek().equals('(')) { stack.pop();}
            else {return false;}
            break;
          case ']':
            if (!stack.empty() && stack.peek().equals('[')) { stack.pop();}
            else {return false;}
            break;
          case '}':
            if (!stack.empty() && stack.peek().equals('{')) { stack.pop();}
            else {return false;}
            break;
      }
    }
    if (stack.empty()) {return true; }
    else {return false; }
  }